### PR TITLE
fix Decay output over LibraryLink

### PR DESCRIPTION
**Pull request by wkotlarski**

### CODE REVIEW

File: templates/info.hpp.in [hunk 1]
```diff
@@ -25,6 +25,8 @@
 #include <array>
 #include <iosfwd>
 #include <string>
+#include <utility>
+#include "boost/optional.hpp"
```
Comment (Expander) [inline templates/info.hpp.in:29]: Should we not write `<boost/optional.hpp>`?

File: meta/FSMathLink.m [hunk 2]
```diff
@@ -311,7 +311,18 @@
                      "n_decays++;\n"
                   ] <>
                   "}\n" <>
-                  "MLPutRule(link, " <> modelName <> "_info::get_particle_name_from_pdg(pid), {\"Decays\"});\n" <>
+                  "auto multiplet_and_index_pair = " <> modelName <> "_info::get_multiplet_and_index_from_pdg(pid);\n" <>
```
Comment (Expander) [inline meta/FSMathLink.m:314]: Can we make this variable `const`?

Comment (wkotlarski) [reply meta/FSMathLink.m:314]: ofc

**Comment by Expander** [comment]
Currently the output format is of the form:
~~~
{CMSSM -> {
   Decays[hh[1]] -> {...}, 
   Decays[hh[2]] -> {...},
   ...
  }
} 
~~~
I'm wondering why there is the `Decays` head? Can we omit it? I think this would be cleaner and it seems unnecessary.

File: templates/info.hpp.in (hunk 1 above)
Comment (wkotlarski) [reply templates/info.hpp.in]: Done.

File: meta/FSMathLink.m (hunk 2 above)
Comment (wkotlarski) [reply meta/FSMathLink.m]: Done.

**Comment by wkotlarski** [comment]
That was the original design from Dylan (I'm not blaming, just stating the fact). I don't have any preference about it. I'll remove it as you ask. If I have a preference regarding something is this printing of a model. Not sure if it's necessary.

**Comment by wkotlarski** [comment]
I've removed the `Decays` head.

**Comment by Expander** [comment]
Thanks!

We should keep the printing of the model, because of EFT towers: There may be multiple models stacked on top of each other. In such a case there must be some way to say in which model a decay has been calculated.

**Comment by Expander** [comment]
Actually I have another wish.

This complicated boolean expression appears twice:
~~~.cpp
         if (!(decays_list.get_total_width() > 0.) || this->get_fd_settings().get(FlexibleDecay_settings::min_br_to_print) > decay.second.get_width()/decays_list.get_total_width()) {
            continue;
         }
~~~
Can you please put it into a local lambda, so we can write
~~~.cpp
auto is_invalid_decay = [&] (...) { ... };

// ...

if (is_invalid_decay(...)) {
   continue;
}
~~~

**Comment by wkotlarski** [comment]
Why lambda and not a variable?

**Comment by Expander** [comment]
Ah, yes, that would be even simpler! :)

**Comment by Expander** [comment]
Thanks a lot, it is very nice now!

**Review by Expander — APPROVED** [review]

**Pull request merged**